includes/shortcodes/class-wc-shortcode-products.php

```diff
@@ -184,7 +184,7 @@ protected function parse_query_args() {
 
 		$orderby_value         = explode( '-', $query_args['orderby'] );
 		$orderby               = esc_attr( $orderby_value[0] );
-		$order                 = ! empty( $orderby_value[1] ) ? $orderby_value[1] : strtoupper( $this->attributes['order'] );
+		$order                 = ! empty( $orderby_value[1] ) ? $orderby_value[1] : ( empty( $_GET['orderby'] ) ? strtoupper( $this->attributes['order'] ) : 'ASC' ); // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
```

Comment:
I think a better solution in terms of readability and future maintenance would be to explicitly check if we are going to use `order` and `orderby` from shortcode or from query param. 
In this case, that would mean we start by getting `order` and `orderby` values from `attributes` and then overwriting them inside an `if` block if query param `orderby` is present. What do you think?

Reply:
Thank you. I've made the nice improvement suggested by you.
Currently I'm using default sorting order of `ASC` when it is not explicitly present in query param, similar to:
https://github.com/woocommerce/woocommerce/blob/8bd6a44763caf22254b2649201560fe9d7564d09/includes/shortcodes/class-wc-shortcode-products.php#L119

I also noticed that, selecting `Sort by latest` in dropdown, I expect the products to be sorted in a way so that newly added products appear at top but because of default `ASC` order it does opposite. 

Am I missing something? Do we need to modify following?
https://github.com/woocommerce/woocommerce/blob/8bd6a44763caf22254b2649201560fe9d7564d09/includes/wc-template-functions.php#L1354-L1364

Reply:
Yeah looks like it. Using `date-desc` makes sense.

Reply:
I think we should make default order `DESC` as most of "Sort by" options are expected to use `DESC` order. For e.g. Sort by rating should show the highest rated products first, similarly sorting by popularity should present most popular products first.

If we use default sorting order of `DESC` when no sort order parameter is present (e.g. `rating` not `rating-desc` etc.) then we just need to flip options for price. We can use `price` in place of `price-desc` and `price-asc` in place of `price`. What do you think?

Reply:
I agree with your suggestion to use `DESC` as default if most of the other attributes are defaulting to DESC anyway.

Other thing is, I think changing the default for price might not be a good idea beause it will break for stores who have hardcoded `orderby=price` in some button or some link. We should probably add a special `if` condition in `includes/shortcodes/class-wc-shortcode-products.php` to make sure that if arg is `price` then we default to `ASC` and also add a comment  explaining why this special condition was required.